parquet-thrift/src/main/java/org/apache/parquet/thrift/struct/CompatibilityChecker.java

```diff
@@ -32,7 +34,7 @@
 import org.apache.parquet.thrift.struct.ThriftType.StringType;
 
 /**
- * A checker for thrift struct, returns compatibility report based on following rules:
+ * A checker for thrift struct to enforce its backward compatibility, returns compatibility report based on following rules:
  * 1. Should not add new REQUIRED field in new thrift struct. Adding optional field is OK
  * 2. Should not change field type for an existing field
  * 3. Should not delete existing field
```

Comment:
Nevermind, this is a forward-compatibility requirement. I don't think this is required for backward-compatibility though. Deleting an existing field will cause the code to ignore it. Things might get weird when you try to add another field with the same ID, but that's a separate issue.